FirebaseDatabase/Sources/Persistence/FLevelDBStorageEngine.m

```diff
@@ -267,11 +267,16 @@ - (void)close {
 }
 
 + (NSString *)firebaseDir {
-#if TARGET_OS_IOS || TARGET_OS_TV
+#if TARGET_OS_IOS || TARGET_OS_WATCH
     NSArray *dirPaths = NSSearchPathForDirectoriesInDomains(
         NSDocumentDirectory, NSUserDomainMask, YES);
     NSString *documentsDir = [dirPaths objectAtIndex:0];
     return [documentsDir stringByAppendingPathComponent:@"firebase"];
+#elif TARGET_OS_TV
+    NSArray *dirPaths = NSSearchPathForDirectoriesInDomains(
+        NSCachesDirectory, NSUserDomainMask, YES);
```

Comment:
Does this change drop the existing cache for Apple TV users?

Reply:
It actually is broken as is - the `NSDocumentsDirectory` is not writable on tvOS devices. Unfortunately it's writeable in simulators so we didn't catch it before. This should work as expected now though!